### PR TITLE
util.h: add the macro ROUNDUP_OVERFLOW()

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Adds the macro ROUNDUP_OVERFLOW() which rounds up like ROUNDUP() but also
checks the result for overflow and returns true on overflow.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

**Comment by jenswi-linaro** [comment]
Tag applied

File: lib/libutils/ext/include/util.h [hunk 1]
```diff
@@ -47,6 +47,14 @@
 #define ROUNDUP(v, size) (((v) + ((__typeof__(v))(size) - 1)) & \
 			  ~((__typeof__(v))(size) - 1))
 
+#define ROUNDUP_OVERFLOW(v, size, res) (__extension__({ \
+	typeof(*res) __roundup_tmp = 0; \
```
Comment (etienne-lms) [inline lib/libutils/ext/include/util.h]: missing parentheses ? `typeof(*(res)) __...`

Comment (jenswi-linaro) [reply lib/libutils/ext/include/util.h]: You're right, I'll fix.

**Comment by jenswi-linaro** [comment]
Addressed comment by @etienne-lms

**Comment by jenswi-linaro** [comment]
Tag applied

**Pull request merged**